File: sdk/java-sdk/src/test/java/com/evernym/verity/sdk/TestBase.java
package com.evernym.verity.sdk;

import com.evernym.verity.sdk.utils.Context;

import static org.junit.jupiter.api.Assertions.fail;

public abstract class TestBase {

    public void withContext(ThrowingConsumer<Context> testCode) throws Exception {
        Context context = null;
        try {
            context = TestHelpers.getContext();
            testCode.accept(context);
        } catch(Exception e) {
            e.printStackTrace();
            fail();
        } finally {
            TestHelpers.cleanup(context);
        }
    }
}

File: sdk/dotnet-sdk/src/VeritySDK.Test/TestBase.cs
using Com.Evernym.Vdrtools.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.Collections.Generic;
using System.Json;
using System.Text;
using VeritySDK.Utils;
using VeritySDK.Wallets;

namespace VeritySDK.Test
{
    public abstract class TestBase
    {
        public delegate void OnContextAction(Context context);

        public void withContext(OnContextAction action)
        {
            Context context = null;
            try
            {
                context = TestHelpers.getContext();
                action(context);
            }
            catch (Exception e)
            {
                Assert.Fail(e.Message);
            }
            finally
            {
                TestHelpers.cleanup(context);
            }
        }

    }
}
